launch_chffrplus.sh

```diff
@@ -64,49 +66,60 @@ function launch {
   echo 0-2 > /dev/cpuset/android/cpus
   echo 0-3 > /dev/cpuset/app/cpus
 
-  # Collect RIL and other possibly long-running I/O interrupts onto CPU 1
-  echo 1 > /proc/irq/78/smp_affinity_list # qcom,smd-modem (LTE radio)
-  echo 1 > /proc/irq/33/smp_affinity_list # ufshcd (flash storage)
-  echo 1 > /proc/irq/35/smp_affinity_list # wifi (wlan_pci)
-  # USB traffic needs realtime handling on cpu 3
-  [ -d "/proc/irq/733" ] && echo 3 > /proc/irq/733/smp_affinity_list # USB for LeEco
-  [ -d "/proc/irq/736" ] && echo 3 > /proc/irq/736/smp_affinity_list # USB for OP3T
-
-  DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
-
-  # Remove old NEOS update file
-  # TODO: move this code to the updater
-  if [ -d /data/neoupdate ]; then
-    rm -rf /data/neoupdate
-  fi
+  # Configure non-default interrupt affinities for NEOS platforms
+  # Slight differences between OP3T and LeEco for I2C, SPS, and USB, trying both for now
+  # TODO: abstract this into per-platform and per-mainboard startup scripts
+  #
+  # Move RIL and other possibly long-running I/O interrupts onto core 1
+  # Move USB to core 3 for better realtime handling
+  IRQ_AFFINE_CORE_0=""
+  IRQ_AFFINE_CORE_1="13 16 25 78"  # I2C, NGD, qcom,smd-modem
+  IRQ_AFFINE_CORE_2="6 15 26 33 35"  # SPS, MDSS, ufshcd (flash storage), wlan_pci (WiFi)
+  IRQ_AFFINE_CORE_3="733 736"  # USB for LeEco and OP3T mainboards respectively
+
+  for CORE in {0..3}
+  do
+    CORE_IRQ_LIST=IRQ_AFFINE_CORE_$CORE
+    for IRQ in ${!CORE_IRQ_LIST}
+    do
+      if [ -d "/proc/irq/$IRQ" ]; then
+        echo "Setting IRQ affinity: IRQ $IRQ to core $CORE"
+        echo $CORE > /proc/irq/$IRQ/smp_affinity_list
+      fi
+    done
+  done
 
   # Check for NEOS update
-  if [ $(< /VERSION) != "14" ]; then
-    if [ -f "$DIR/scripts/continue.sh" ]; then
-      cp "$DIR/scripts/continue.sh" "/data/data/com.termux/files/continue.sh"
+  if [ "$(< /VERSION)" != "$NEOS_VERSION" ]; then
+    if [ -f "$BASEDIR/scripts/continue.sh" ]; then
+      cp "$BASEDIR/scripts/continue.sh" "/data/data/com.termux/files/continue.sh"
     fi
 
     if [ ! -f "$BASEDIR/prebuilt" ]; then
       echo "Clearing build products and resetting scons state prior to NEOS update"
-      cd $BASEDIR && scons --clean
+      git clean -xdf
```

Comment:
Why no more scons clean? Git clean -xdf doesn't clean the submodules.

Reply:
This bit of script was about downgrades, not upgrades (#1677). I had streamlined it a bit since scons clean isn't effective in that case anyway, and it added several seconds of unusual black-screen UI between the bootup spinner and the NEOS update dialog. I took out the explicit database rm since git clean -xdf does it.

I haven't yet noticed an issue with scons needing help to handle upgrades correctly, even without cleanup. However, your point is well taken about submodules. Even though I think scons rebuilds everything anyway after the database zap, there's benefit to a complete clean. I'll do so after the database zap so it works in both upgrade and downgrade cases.

Reply:
You can also add a `git submodule foreach --recursive git clean -xdf` to clean the submodules.

Reply:
That seems preferable to the heavyweight scons clean then.